### ISSUE TITLE
Allow matchers inside objects

### ISSUE THREAD

**Issue by dmcquay** [testdouble/testdouble.js#160]
If I have a function that takes parameters as an options object, like so:

```js
// foo.js
module.exports = {
  function bar({a, b}) {
    doSomething() 
  }
}
```

I would like to be able to do this:

```js
td.replace(foo, 'bar')
td.when(
  foo.bar({
    a: td.matchers.isA(Date),
    b: 'hello world'
  })
).thenReturn('happy days')
```

**Comment by searls** [comment]
This should be resolved by #142 — I think

**Comment by dmcquay** [comment]
Indeed! And I see you got that issue moving today. Thanks!

**Comment by searls** [comment]
This landed in #203 and is in the latest 2.0 prerelease
